src/Symfony/Component/Mailer/Transport/Smtp/Stream/SocketStream.php

```diff
@@ -26,11 +26,16 @@ final class SocketStream extends AbstractStream
     private $url;
     private $host = 'localhost';
     private $port = 465;
-    private $timeout = 5;
+    private $timeout;
     private $tls = true;
     private $sourceIp;
     private $streamContextOptions = [];
 
+    public function __construct()
+    {
+        $this->timeout = (float) ini_get('default_socket_timeout');
```

Comment:
`return $this->timeout ?? (float) ini_get('default_socket_timeout');` in getTimeout would be more flexible I think